### ISSUE TITLE
[FEA] DGL 1.0 support

### ISSUE THREAD

**Issue by lmeyerov** [graphistry/pygraphistry#434]
**Is your feature request related to a problem? Please describe.**

We are upgrading base images, and are trying to decide which version of DGL to support

**Describe the solution you'd like**

* Test matrix support of DGL methods across intended versions, including 1.x and lowest-supported <1.0
* Version support guidance in the docs

**Describe alternatives you've considered**

Sticking with < 1.0 or doing only >= 1.0

**Additional context**

Came up when `make base_native` failed on DGL < 1.0

**Comment by lmeyerov** [comment]
cc + @exrhizo @silkspace

**Comment by jrmendeshurb4** [comment]
ok

**Comment by jrmendeshurb4** [comment]
ok

**Comment by lmeyerov** [comment]
@tanmoyio @silkspace thoughts? Our new docker base image uses DGL 1.0, not sure pygraphistry [ai] supports?

**Comment by tanmoyio** [comment]
@lmeyerov I just ran all the `dgl` dependent tests with `dgl-1.0.0`
`graphistry/tests/test_dgl_utils.py`
`graphistry/tests/test_embed_utils.py`
all the tests are passing. 

Also I went through `dgl-1.0.0` change logs I don't think they introduced significant changes in the core functionality which will affect us. Although I have found some interesting stuffs. 

They added - 
- Support canonical edge types in HeteroGraphConv(this is being used by our `.embed()`)
- Support bfloat16
